cli/src/doc.rs

```diff
@@ -34,66 +40,85 @@ impl fmt::Display for DocFormat {
     }
 }
 
-pub fn export_doc(
-    program: &mut Program<CacheImpl>,
-    file: Option<&PathBuf>,
-    output: Option<PathBuf>,
-    stdout: bool,
-    format: DocFormat,
-) -> Result<(), Error> {
-    let doc = program.extract_doc()?;
-    let mut out: Box<dyn std::io::Write> = if stdout {
-        Box::new(std::io::stdout())
-    } else {
-        Box::new(
-            output
-                .as_ref()
-                .map(|output| {
-                    fs::File::create(output.clone()).map_err(|e| {
-                        Error::IOError(IOError(format!(
-                            "when opening or creating output file `{}`: {}",
-                            output.to_string_lossy(),
-                            e
-                        )))
+#[derive(clap::Parser, Debug)]
+pub struct DocOptions {
+    /// The path of the generated documentation file. Default to
+    /// `~/.nickel/doc/<input-file>.md` for input `<input-file>.ncl`, or to
+    /// `~/.nickel/doc/out.md` if the input is read from stdin.
+    #[arg(short, long)]
+    pub output: Option<PathBuf>,
+    /// Write documentation to stdout. Takes precedence over `output`
+    #[arg(long)]
+    pub stdout: bool,
+    /// The output format for the generated documentation.
+    #[arg(long, value_enum, default_value_t)]
+    pub format: crate::doc::DocFormat,
+
+    #[command(flatten)]
+    pub sources: Files,
+}
+
+impl DocOptions {
+    pub fn run(self, global: GlobalOptions) -> CliResult<()> {
```

Comment:
It's maybe not very important, but I find it a bit strange conceptually to implement a `run` method on something called `XXXOptions`, which I would expect to be pure data. I could see either free-standing functions in each related module. On the other hand, I can see how the `XXXOptions` are also subtypes of the `Command` type morally, so maybe it's just a naming thing after all. Maybe we should name them `XXXSubcommand` or something? I don't have a strong opinion either, so if you think this is best as it is, fine by me.

Reply:
A freestanding `run` felt worse to me, although my reasons aren't really well defined. Ultimately I agree this is really an issue with the naming of the `XXXOptions` and I changed them to `XXXCommand`.